### DISCUSSION TITLE
Integration of Builderius with popular themes

### DISCUSSION THREAD

**Discussion by thisbit** [builderius/builderius#35] [Ideas]
Builderius does not dissable a theme and this is good. You can use the theme and then use bilderius to develop a special page. But for now, it has to be an entire page. Both for better Builderius adoption and for general flexibility/productivity it would be great to be able to design only a part of the page.

Some ideas on the implementation:

### GeneratePress Premium (free version is 500 000+ active installations), Kadence Pro (free version is 100 000+ active installations), Blocksy Pro (free version is 90 000+ active installations) and Astra Pro (free version is 10 00 000+ active installations)

GP has a module called [Elements](https://docs.generatepress.com/collection/elements/), this is A custom post type which can be assigned a location (similar to Builderius templates) like, single, archive ... and a [template_hook](https://docs.generatepress.com/article/hooks-visual-guide/) which is the location on the said pages.
This makes "Elements" template parts with conditional display. However, not everything can designed well with Blocks, so for instance custom menus, cannot. And In this case if Builderius would work with this, one could design the element with builderius, and inject it where needed accross the theme.

All the themes in the title have this feature in some shape:
- [Astra](https://wpastra.com/docs/custom-layout-hooks/)
- [Blocksy](https://creativethemes.com/blocksy/premium/#features), scroll to Content Blocks
- [Kadence](https://www.kadencewp.com/kadence-theme/premium/), also scroll down to Hooked elements
- [GeneratePress](https://docs.generatepress.com/collection/elements/), they have this most advanced, and public docs

Other builders also use these features of the themes. Because users often do not want to design the entire site, they use the theme, and then jsut add features they need with a builder.

I would use it for these kind of things on productions sites right now, for isntance.

**Comment by seaj1one** [comment]
I was planning on building each page using template parts via Metabox. To dynamically create content for each section. 

What's the advantage to use the theme in general over ACF or Metabox?

Is the theme for clients the only way or easiest way for non-developers for clients to add content to their website?

**Reply by mrpsiho** [reply]
It is not clear what and how you would use it. Could you clarify?

**Comment by mrpsiho** [comment]
"...if Builderius would work with this, one could design the element with builderius, and inject it where needed accross the theme".

Now: Builderius template parts can be added to any Builderius templates
Planned functionality: Builderius tmpl parts via hooks could be added anywhere via standard and custom WP hooks. More specifically - actions (because filters are not meant for visual parts, basically). So, we will integrate not with specific theme or themes, but with WP hooks system in general. https://developer.wordpress.org/plugins/hooks/

**Comment by seaj1one** [comment]
Should I reply @mrpsiho ?

**Reply by mrpsiho** [reply]
Yes, reply pls.

**Comment by seaj1one** [comment]
coolio - what's the advantage of using theme hooks and theme blocks a good thing over ACF or Metabox?

**Comment by seaj1one** [comment]
@thisbit I didn't see the benefit but after the latest release, I now know.

**Reply by thisbit** [reply]
Thanks for saying this. 😁
